File: assignments/assign01/assn01_ws/src/heatControl_py/heatControl_py/heater.py
#!/usr/bin/env python3
import rclpy   # rclpy must be declared as a dependence in package.xml - this should be added automatically when you create the package
from rclpy.node import Node

from example_interfaces.msg import String

class Heater(Node):
    def __init__(self):
        super().__init__("heater")

        # create subscriber
        self.heaterSubscriber_ = self.create_subscription(
            String,
            "fanCmd",
            self.callback_handleCmd,
            10
        )

        self.initialize()

    def callback_handleCmd(self, msg):
        self.cmd_ = msg.data
        self.get_logger().info("the received cmd is " + self.cmd_)

    def initialize(self):
        self.cmd_ = ""

def main(args = None):
    rclpy.init(args=args)
    node = Heater()
    rclpy.spin(node)
    rclpy.shutdown()        
        
if __name__ == "__main__":
    main()  

File: assignments/assign01/assn01_ws/src/heatControl_cpp/src/heater.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class Heater: public rclcpp::Node
{
public:
   Heater() : Node("heater"), cmd_("")
   {
        heaterSubscriber_ = this->create_subscription<example_interfaces::msg::String>(
            "fanCmd", 
            10, 
            std::bind(&Heater::callback_handleCmd, this, std::placeholders::_1)
        );
   }

   void callback_handleCmd(const example_interfaces::msg::String::SharedPtr msg)
   {
        cmd_ = msg->data;
        RCLCPP_INFO(this->get_logger(), "the received cmd is %s\n", cmd_.c_str());
   }

private:
   std::string cmd_;
   rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr heaterSubscriber_;
};


int main(int argc, char **argv)
{
   rclcpp::init(argc, argv);
   auto node = std::make_shared<Heater>();
   rclcpp::spin(node);
   rclcpp::shutdown();
   return 0;
}